it/src/main/java/com/google/cloud/teleport/it/datastream/DatastreamResourceManager.java

```diff
@@ -0,0 +1,30 @@
+
+package com.google.cloud.teleport.it.datastream;
+
+import com.google.cloud.datastream.v1.ConnectionProfile;
+import com.google.cloud.datastream.v1.Stream;
+
+import java.util.concurrent.ExecutionException;
+
+/**
+ * Interface for managing datastream resources in integration tests.
+ */
+public interface DatastreamResourceManager {
+    ConnectionProfile createSourceConnectionProfile(String connectionProfileId)
+            throws ExecutionException, InterruptedException;
+
+    ConnectionProfile createDestinationConnectionProfile(String connectionProfileId)
+            throws ExecutionException, InterruptedException;
+
+    Stream createStream(String streamId,
+                        String sourceConnectionProfile,
+                        String destinationConnectionProfile)
+            throws ExecutionException, InterruptedException;
+
+    Stream startStream(String streamId) throws ExecutionException, InterruptedException;
+
+    Stream pauseStream(String streamID) throws ExecutionException, InterruptedException;
+
+    void cleanUpAll() throws ExecutionException, InterruptedException;
```

Comment:
This line can be removed since `cleanupAll()` is inherited. This is so that when we run a suite of integration tests, we can pass multiple resource managers that were used for the tests in a cleanup function that treats all of them as an instance of `ResourceManager` which is guaranteed to have a `cleanupAll()` method.
```suggestion
```

Reply:
done